api.planx.uk/inviteToPay/sendPaymentEmail.ts

```diff
@@ -1,37 +1,30 @@
+import { PaymentRequest } from '@opensystemslab/planx-core';
 import { gql } from "graphql-request";
 import { Template, calculateExpiryDate, convertSlugToName, getClientForTemplate, getHumanReadableProjectType, sendEmail } from "../saveAndReturn/utils";
 import { InviteToPayNotifyConfig } from "../types";
+import { Team } from '../types';
+
+interface SessionDetails {
+  email: string;
+  flow: {
+    slug: string;
+    team: Team;
+  }
+};
+
+interface ValidatePaymentRequest {
+  query: PaymentRequest & { session: SessionDetails }
+};
 
 const sendSinglePaymentEmail = async (
   template: Template,
   paymentRequestId: string,
 ) => {
   try {
-    const payment = await validatePaymentRequest(paymentRequestId, template);
-    if (!payment) throw Error("Unable to send payment email, cannot find payment request");
-    
-    const humanReadableProjectType = await getHumanReadableProjectType(payment.session_preview_data);
-    const config: InviteToPayNotifyConfig = {
-      personalisation: {
-        id: payment.session_id,
-        serviceName: convertSlugToName(payment.session.flow.slug),
-        emailReplyToId: payment.session.flow.team.notify_personalisation.emailReplyToId,
-        helpEmail: payment.session.flow.team.notify_personalisation.helpEmail,
-        helpOpeningHours: payment.session.flow.team.notify_personalisation.helpOpeningHours,
-        helpPhone: payment.session.flow.team.notify_personalisation.helpPhone,
-        paymentRequestId: paymentRequestId,
-        payeeEmail: payment.payee_email,
-        payeeName: payment.payee_name,
-        agentName: payment.applicant_name,
-        paymentLink: `https://${payment.session.flow.team.domain}/${payment.session.flow.slug}/pay?paymentRequestId=${paymentRequestId}`,
-        fee: Number(payment.payment_amount).toLocaleString("en-GB", { style: "currency", currency: "GBP" }),
-        projectType: humanReadableProjectType || "Project type not submitted",
-        address: payment.session_preview_data?._address.title,
-        expiryDate: calculateExpiryDate(payment.created_at),
-      }
-    };
-
-    const recipient = template.includes("-agent") ? payment.session.email : payment.payee_email;
+    const { session, paymentRequest } = await validatePaymentRequest(paymentRequestId, template);
+    if (!session || !paymentRequest || Boolean(paymentRequest?.paidAt)) throw Error(`Invalid payment request: ${paymentRequestId}`);
```

Comment:
I think we can remove `... || Boolean(paymentRequest?.paidAt` here since after rebasing it's also below on L72 in the validate method and will throw a more specific "already paid" error.

Reply:
Thank you - missed this in the rebase.